### PR TITLE
Implemented more performant deep clone helper

### PR DESCRIPTION
**Pull request by sunpietro**
This is another pack of code improvements. The behaviour of Universal Discovery should not differ from the behaviour on master.

### CODE REVIEW

File: src/modules/common/helpers/deep.clone.helper.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Clones any object. Faster alternative to `JSON.parse(JSON.stringify)`
+ *
+ * @function deepClone
+ * @param {Any} data
+ * @returns {Any} cloned data
+ */
+const deepClone = (data) => {
```
Comment (tischsoic) [inline src/modules/common/helpers/deep.clone.helper.js:8]: Did you make some speed tests? I have made few simple tests and it seems that the speeds are basicaly the same.
But these were quite simple speed tests. :)

Comment (tischsoic) [reply src/modules/common/helpers/deep.clone.helper.js:8]: My simple test suite: https://jsbin.com/zetanapoha/edit?js,output 🏎 :)

Comment (sunpietro) [reply src/modules/common/helpers/deep.clone.helper.js:8]: This is the same approach we're using in Page Builder. I wanted to make it inline.

Comment (tischsoic) [reply src/modules/common/helpers/deep.clone.helper.js:8]: > inline

I don't know what you mean by that?

Comment (tischsoic) [reply src/modules/common/helpers/deep.clone.helper.js:8]: It is just to make it behave the same way as in Page Builder?

Comment (sunpietro) [reply src/modules/common/helpers/deep.clone.helper.js:8]: Exactly

Comment (tischsoic) [reply src/modules/common/helpers/deep.clone.helper.js:8]: I think that deep clone output should always be the same, why not use previous approach from UDW in Page Builder? It is much simpler and it seems that performance is the same. But nevermind. :)

Comment (sunpietro) [reply src/modules/common/helpers/deep.clone.helper.js:8]: I did other performance tests and it told me that the previous approach was faster only in FF. In other browsers this approach was better.
We can discuss it in private and, eventually, prepare test cases for each option :)

**Comment by sunpietro** [comment]
This branch has been rebased with master

File: src/modules/common/helpers/deep.clone.helper.js (hunk 1 above)
Comment (tischsoic) [reply src/modules/common/helpers/deep.clone.helper.js:8]: If you performed more detailed tests and your approach performed better then OK. :)
Maybe in the future will find some time to test it more thoroughly. :P

**Review by tischsoic — APPROVED** [review]

**Review by barbaragr — APPROVED** [review]

**Pull request merged**